aiomcache/client.py

```diff
@@ -148,9 +173,10 @@ async def get(self, conn: Connection, key: bytes, default: bytes) -> bytes:
     @acquire
     async def get(
         self, conn: Connection, key: bytes, default: Optional[bytes] = None
-    ) -> Optional[bytes]:
+    ) -> Optional[Any]:
```

Comment:
We should have a quick think over whether there's a way to keep some useful type information. If we can't get types when using the handlers, maybe we can atleast make the class Generic, so when the handler is None it will still return bytes.

Reply:
that won't help much for the flag case, in that scenario it will still need to be generic

Reply:
Yes, but flags is a new feature. I'm just interested in avoiding a regression for current users who do not use flags.

Reply:
In fact, isn't the return type here essentially the return type of the handler? So, we could potentially do something useful by setting the handler's return type to a TypeVar and then set the return type here to `Union[bytes, _T]`...

With just a plain `bytes` when handler is `None`.

Reply:
added some typing changes, let me know what you think

Reply:
hmm, let me know if you know how to fix the mypy issues

Reply:
So, e.g. the first error, is because it contains a generic type (because that type includes a TypeVar), so any reference to it must declare that type. It can use another TypeVar, like `_ClientT[_T]`.

I'll play with it over the weekend and see if I can get the typing working, it's fairly complex.

Reply:
OK, I think a good option is to use separate classes, where the original `Client` can be defined like so:
```
class Client(FlagClient[bytes]):
    def __init__(self, host: str, port: int = 11211, *,
                 pool_size: int = 2, pool_minsize: Optional[int] = None):
        super().__init__(host, port, pool_size=pool_size, pool_minsize=pool_minsize,
                         get_flag_handler=None, set_flag_handler=None)
```

Any issues with that? Also, should it be possible to use only the get or set handler, or should they both be required? And, is there any reason that the handlers might want different types (i.e. `GetHandler[_T]` and `GetHandler[_U]`, rather than both being `_T`)?

I'll come back to finish the implementation next week, as I need support for `Concatenate` which will only be in the next release of mypy.

Reply:
ya that sounds fine to have two clients.  Technically you can have either handler, not sure the use cases of having just one though.  Unless there's a use case I say we require both.  I think they're different types right as one accepts any type and converts to bytes, and the other vice-versa.  Have an example?